{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 데이터 불러오기"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3325, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-c9d8de687324>\", line 1, in <module>\n",
      "    from tensorflow.keras.datasets import imdb\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 101, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\__init__.py\", line 40, in <module>\n",
      "    from tensorflow.python.tools import module_util as _module_util\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2039, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 319, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 353, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 967, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 677, in _load_unlocked\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\__init__.py\", line 42, in <module>\n",
      "    from . _api.v2 import audio\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\_api\\v2\\audio\\__init__.py\", line 10, in <module>\n",
      "    from tensorflow.python.ops.gen_audio_ops import decode_wav\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\ops\\gen_audio_ops.py\", line 9, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow as _pywrap_tensorflow\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3325, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-c9d8de687324>\", line 1, in <module>\n",
      "    from tensorflow.keras.datasets import imdb\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 101, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\__init__.py\", line 40, in <module>\n",
      "    from tensorflow.python.tools import module_util as _module_util\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2039, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3325, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-c9d8de687324>\", line 1, in <module>\n",
      "    from tensorflow.keras.datasets import imdb\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 101, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\__init__.py\", line 40, in <module>\n",
      "    from tensorflow.python.tools import module_util as _module_util\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2039, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3248, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3342, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2042, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1385, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1288, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in structured_traceback\n",
      "    formatted_exceptions += self.prepare_chained_exception_message(evalue.__cause__)\n",
      "TypeError: can only concatenate str (not \"list\") to str\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2039, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 319, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 353, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 967, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 677, in _load_unlocked\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\__init__.py\", line 42, in <module>\n",
      "    from . _api.v2 import audio\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\_api\\v2\\audio\\__init__.py\", line 10, in <module>\n",
      "    from tensorflow.python.ops.gen_audio_ops import decode_wav\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\ops\\gen_audio_ops.py\", line 9, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow as _pywrap_tensorflow\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3325, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-c9d8de687324>\", line 1, in <module>\n",
      "    from tensorflow.keras.datasets import imdb\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 101, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\__init__.py\", line 40, in <module>\n",
      "    from tensorflow.python.tools import module_util as _module_util\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2039, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3248, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3342, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2042, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1385, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1288, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in structured_traceback\n",
      "    formatted_exceptions += self.prepare_chained_exception_message(evalue.__cause__)\n",
      "TypeError: can only concatenate str (not \"list\") to str\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2039, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 242, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"C:\\Users\\jcy12\\Anaconda3\\lib\\imp.py\", line 342, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: 지정된 모듈을 찾을 수 없습니다.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "can only concatenate str (not \"list\") to str",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mImportError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     57\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 58\u001b[1;33m   \u001b[1;32mfrom\u001b[0m \u001b[0mtensorflow\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpython\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpywrap_tensorflow_internal\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[1;33m*\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     59\u001b[0m   \u001b[1;32mfrom\u001b[0m \u001b[0mtensorflow\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpython\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpywrap_tensorflow_internal\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0m__version__\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     27\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0m_mod\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 28\u001b[1;33m     \u001b[0m_pywrap_tensorflow_internal\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mswig_import_helper\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     29\u001b[0m     \u001b[1;32mdel\u001b[0m \u001b[0mswig_import_helper\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\u001b[0m in \u001b[0;36mswig_import_helper\u001b[1;34m()\u001b[0m\n\u001b[0;32m     23\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 24\u001b[1;33m                 \u001b[0m_mod\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mimp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mload_module\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'_pywrap_tensorflow_internal'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpathname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdescription\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     25\u001b[0m             \u001b[1;32mfinally\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\imp.py\u001b[0m in \u001b[0;36mload_module\u001b[1;34m(name, file, filename, details)\u001b[0m\n\u001b[0;32m    241\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 242\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mload_dynamic\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfilename\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfile\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    243\u001b[0m     \u001b[1;32melif\u001b[0m \u001b[0mtype_\u001b[0m \u001b[1;33m==\u001b[0m \u001b[0mPKG_DIRECTORY\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\imp.py\u001b[0m in \u001b[0;36mload_dynamic\u001b[1;34m(name, path, file)\u001b[0m\n\u001b[0;32m    341\u001b[0m             name=name, loader=loader, origin=path)\n\u001b[1;32m--> 342\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0m_load\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mspec\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    343\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mImportError\u001b[0m: DLL load failed: 지정된 모듈을 찾을 수 없습니다.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\u001b[0m in \u001b[0;36mshowtraceback\u001b[1;34m(self, exc_tuple, filename, tb_offset, exception_only, running_compiled_code)\u001b[0m\n\u001b[0;32m   2038\u001b[0m                         \u001b[1;31m# in the engines. This should return a list of strings.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2039\u001b[1;33m                         \u001b[0mstb\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_render_traceback_\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2040\u001b[0m                     \u001b[1;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'ImportError' object has no attribute '_render_traceback_'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\u001b[0m in \u001b[0;36mrun_code\u001b[1;34m(self, code_obj, result, async_)\u001b[0m\n\u001b[0;32m   3340\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   3341\u001b[0m                 \u001b[0mresult\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0merror_in_exec\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msys\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mexc_info\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 3342\u001b[1;33m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshowtraceback\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrunning_compiled_code\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   3343\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   3344\u001b[0m             \u001b[0moutflag\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mFalse\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\u001b[0m in \u001b[0;36mshowtraceback\u001b[1;34m(self, exc_tuple, filename, tb_offset, exception_only, running_compiled_code)\u001b[0m\n\u001b[0;32m   2040\u001b[0m                     \u001b[1;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2041\u001b[0m                         stb = self.InteractiveTB.structured_traceback(etype,\n\u001b[1;32m-> 2042\u001b[1;33m                                             value, tb, tb_offset=tb_offset)\n\u001b[0m\u001b[0;32m   2043\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2044\u001b[0m                     \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_showtraceback\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0metype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mstb\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\u001b[0m in \u001b[0;36mstructured_traceback\u001b[1;34m(self, etype, value, tb, tb_offset, number_of_lines_of_context)\u001b[0m\n\u001b[0;32m   1383\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtb\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtb\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1384\u001b[0m         return FormattedTB.structured_traceback(\n\u001b[1;32m-> 1385\u001b[1;33m             self, etype, value, tb, tb_offset, number_of_lines_of_context)\n\u001b[0m\u001b[0;32m   1386\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1387\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\u001b[0m in \u001b[0;36mstructured_traceback\u001b[1;34m(self, etype, value, tb, tb_offset, number_of_lines_of_context)\u001b[0m\n\u001b[0;32m   1286\u001b[0m             \u001b[1;31m# Verbose modes need a full traceback\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1287\u001b[0m             return VerboseTB.structured_traceback(\n\u001b[1;32m-> 1288\u001b[1;33m                 \u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0metype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtb\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtb_offset\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnumber_of_lines_of_context\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1289\u001b[0m             )\n\u001b[0;32m   1290\u001b[0m         \u001b[1;32melif\u001b[0m \u001b[0mmode\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m'Minimal'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\IPython\\core\\ultratb.py\u001b[0m in \u001b[0;36mstructured_traceback\u001b[1;34m(self, etype, evalue, etb, tb_offset, number_of_lines_of_context)\u001b[0m\n\u001b[0;32m   1148\u001b[0m         \u001b[0mexception\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_parts_of_chained_exception\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mevalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1149\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mexception\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1150\u001b[1;33m             \u001b[0mformatted_exceptions\u001b[0m \u001b[1;33m+=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mprepare_chained_exception_message\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mevalue\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__cause__\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1151\u001b[0m             \u001b[0metype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mevalue\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0metb\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mexception\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1152\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m: can only concatenate str (not \"list\") to str"
     ]
    }
   ],
   "source": [
    "from tensorflow.keras.datasets import imdb\n",
    "\n",
    "num_words = 10000\n",
    "(X_train, y_train), (X_test, y_test) = imdb.load_data(num_words=num_words)\n",
    "\n",
    "from tensorflow.keras.preprocessing.sequence import pad_sequences\n",
    "\n",
    "max_len = 500\n",
    "\n",
    "pad_X_train = pad_sequences(X_train, maxlen=max_len)\n",
    "pad_X_test = pad_sequences(X_test, maxlen=max_len)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 모델 구성하기 - Conv1D"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tensorflow.keras.models import Sequential\n",
    "from tensorflow.keras.layers import Dense, Embedding, Conv1D, MaxPooling1D, GlobalMaxPooling1D\n",
    "\n",
    "model = Sequential()\n",
    "# 이 층은 모델의 제일 첫 번째 층으로만 사용할 수 있습니다.\n",
    "# Flatten 층을 사용하기 위해 input_length를 전달합니다.\n",
    "model.add(Embedding(input_dim = num_words, output_dim = 32, input_length = max_len))\n",
    "model.add(Conv1D(32, 7, activation = 'relu'))\n",
    "model.add(MaxPooling1D(7))\n",
    "model.add(Conv1D(32, 5, activation = 'relu'))\n",
    "model.add(MaxPooling1D(5))\n",
    "model.add(GlobalMaxPooling1D())\n",
    "model.add(Dense(1, activation = 'sigmoid'))\n",
    "\n",
    "model.compile(optimizer='adam',\n",
    "             loss = 'binary_crossentropy',\n",
    "             metrics = ['acc'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 모델 학습시키기"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 20000 samples, validate on 5000 samples\n",
      "Epoch 1/30\n",
      "20000/20000 [==============================] - 12s 625us/sample - loss: 0.4685 - acc: 0.7483 - val_loss: 0.3311 - val_acc: 0.8594\n",
      "Epoch 2/30\n",
      "20000/20000 [==============================] - 8s 396us/sample - loss: 0.2350 - acc: 0.9072 - val_loss: 0.3163 - val_acc: 0.8708\n",
      "Epoch 3/30\n",
      "20000/20000 [==============================] - 7s 373us/sample - loss: 0.1325 - acc: 0.9549 - val_loss: 0.3859 - val_acc: 0.8612\n",
      "Epoch 4/30\n",
      "20000/20000 [==============================] - 8s 392us/sample - loss: 0.0623 - acc: 0.9811 - val_loss: 0.4761 - val_acc: 0.8630\n",
      "Epoch 5/30\n",
      "20000/20000 [==============================] - 8s 383us/sample - loss: 0.0216 - acc: 0.9952 - val_loss: 0.6084 - val_acc: 0.8624\n",
      "Epoch 6/30\n",
      "20000/20000 [==============================] - 7s 358us/sample - loss: 0.0066 - acc: 0.9992 - val_loss: 0.7095 - val_acc: 0.8634\n",
      "Epoch 7/30\n",
      "20000/20000 [==============================] - 7s 357us/sample - loss: 0.0029 - acc: 0.9997 - val_loss: 0.7645 - val_acc: 0.8634\n",
      "Epoch 8/30\n",
      "20000/20000 [==============================] - 8s 377us/sample - loss: 0.0013 - acc: 0.9999 - val_loss: 0.8400 - val_acc: 0.8622\n",
      "Epoch 9/30\n",
      "20000/20000 [==============================] - 7s 372us/sample - loss: 4.3883e-04 - acc: 1.0000 - val_loss: 0.8929 - val_acc: 0.8618\n",
      "Epoch 10/30\n",
      "20000/20000 [==============================] - 7s 357us/sample - loss: 2.9625e-04 - acc: 1.0000 - val_loss: 0.9317 - val_acc: 0.8622\n",
      "Epoch 11/30\n",
      "20000/20000 [==============================] - 7s 359us/sample - loss: 2.4104e-04 - acc: 1.0000 - val_loss: 0.9692 - val_acc: 0.8608\n",
      "Epoch 12/30\n",
      "20000/20000 [==============================] - 7s 357us/sample - loss: 2.0158e-04 - acc: 1.0000 - val_loss: 1.0058 - val_acc: 0.8618\n",
      "Epoch 13/30\n",
      "20000/20000 [==============================] - 7s 360us/sample - loss: 1.6573e-04 - acc: 1.0000 - val_loss: 1.0430 - val_acc: 0.8608\n",
      "Epoch 14/30\n",
      "20000/20000 [==============================] - 7s 357us/sample - loss: 2.5800e-04 - acc: 0.9999 - val_loss: 1.1146 - val_acc: 0.8576\n",
      "Epoch 15/30\n",
      "20000/20000 [==============================] - 7s 353us/sample - loss: 0.0681 - acc: 0.9775 - val_loss: 0.7757 - val_acc: 0.8546\n",
      "Epoch 16/30\n",
      "20000/20000 [==============================] - 7s 355us/sample - loss: 0.0129 - acc: 0.9959 - val_loss: 0.8658 - val_acc: 0.8608\n",
      "Epoch 17/30\n",
      "20000/20000 [==============================] - 7s 354us/sample - loss: 0.0012 - acc: 0.9999 - val_loss: 0.9099 - val_acc: 0.8602\n",
      "Epoch 18/30\n",
      "20000/20000 [==============================] - 7s 353us/sample - loss: 3.4089e-04 - acc: 1.0000 - val_loss: 0.9445 - val_acc: 0.8618\n",
      "Epoch 19/30\n",
      "20000/20000 [==============================] - 7s 355us/sample - loss: 2.4366e-04 - acc: 1.0000 - val_loss: 0.9694 - val_acc: 0.8618\n",
      "Epoch 20/30\n",
      "20000/20000 [==============================] - 7s 355us/sample - loss: 1.9650e-04 - acc: 1.0000 - val_loss: 0.9957 - val_acc: 0.8624\n",
      "Epoch 21/30\n",
      "20000/20000 [==============================] - 7s 355us/sample - loss: 1.6280e-04 - acc: 1.0000 - val_loss: 1.0214 - val_acc: 0.8614\n",
      "Epoch 22/30\n",
      "20000/20000 [==============================] - 7s 355us/sample - loss: 1.3629e-04 - acc: 1.0000 - val_loss: 1.0494 - val_acc: 0.8618\n",
      "Epoch 23/30\n",
      "20000/20000 [==============================] - 8s 385us/sample - loss: 1.1577e-04 - acc: 1.0000 - val_loss: 1.0750 - val_acc: 0.8612\n",
      "Epoch 24/30\n",
      "20000/20000 [==============================] - 8s 390us/sample - loss: 9.8489e-05 - acc: 1.0000 - val_loss: 1.1066 - val_acc: 0.8618\n",
      "Epoch 25/30\n",
      "20000/20000 [==============================] - 8s 408us/sample - loss: 8.3414e-05 - acc: 1.0000 - val_loss: 1.1371 - val_acc: 0.8614\n",
      "Epoch 26/30\n",
      "20000/20000 [==============================] - 8s 379us/sample - loss: 7.0461e-05 - acc: 1.0000 - val_loss: 1.1664 - val_acc: 0.8608\n",
      "Epoch 27/30\n",
      "20000/20000 [==============================] - 9s 443us/sample - loss: 5.8464e-05 - acc: 1.0000 - val_loss: 1.1982 - val_acc: 0.8608\n",
      "Epoch 28/30\n",
      "20000/20000 [==============================] - 8s 388us/sample - loss: 4.7711e-05 - acc: 1.0000 - val_loss: 1.2294 - val_acc: 0.8610\n",
      "Epoch 29/30\n",
      "20000/20000 [==============================] - 7s 365us/sample - loss: 3.8474e-05 - acc: 1.0000 - val_loss: 1.2624 - val_acc: 0.8610\n",
      "Epoch 30/30\n",
      "20000/20000 [==============================] - 7s 367us/sample - loss: 3.0846e-05 - acc: 1.0000 - val_loss: 1.2963 - val_acc: 0.8606\n"
     ]
    }
   ],
   "source": [
    "history = model.fit(pad_X_train, y_train, \n",
    "                    batch_size = 32, epochs = 30, \n",
    "                    validation_split = 0.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmEAAAFNCAYAAABIc7ibAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjAsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+17YcXAAAgAElEQVR4nOzdd3zV1f3H8dcnISQQ9pANQdlLqIADB1WLolasAxeuahFXrVtrHdVqbbVq6wC1VevW6k/BilsRLaAEBGXIlBFQQWYYISQ5vz/OjYSQcUPuzffeb97Px+M+7r3f+UkCN++c7/meY845RERERKRmpQRdgIiIiEhtpBAmIiIiEgCFMBEREZEAKISJiIiIBEAhTERERCQACmEiIiIiAVAIk0qZ2TgzuyUB6jjfzD6Lw3FvN7Pnylk31MxyYn1OEalZtflzTBJXnaALkPgys2XARc65D/b2GM65MbGrSESkavQ5JmGllrBazswUxEUkqelzTJKVQliImdmzQEfgTTPbYmbXm1mWmTkzu9DMVgAfRbb9j5l9b2abzGyymfUucZynzexPkddDzSzHzK4xszVm9p2ZXVBBDReY2XwzyzWzpWZ2cYl1FR7LzJqb2QQz22xmXwD7VXCed8zs8lLLZpvZyZHXfzezlZFjzTCzw6r6/Ywcp6eZTTKzjWY218xOLLHuODObF/laV5nZtZHlLczsv5F91pvZp2am/3siUdDnWPU/x8ysaeQzaK2ZbYi8bl9ifTMze8rMVkfWv1Fi3QgzmxU55xIzOzaac0p09IsgxJxz5wArgF865xo45/5aYvURQE/gmMj7t4GuwD7ATOD5Cg7dGmgMtAMuBB4xs6blbLsGOAFoBFwAPGBmP4vyWI8AeUAb4NeRR3leAM4sfmNmvYBOwFuRRdOB/kCzyLb/MbOMCo63BzNLA94E3sN/n64Anjez7pFN/gVc7JxrCPQh8osBuAbIAVoCrYDfA5ovTCQK+hyLyedYCvBU5Fgdge3AwyXWPwvUB3rjv3cPRM4/GHgGuA5oAhwOLIvifBIt55weIX7g/8McXeJ9Fj4A7FvBPk0i2zSOvH8a+FPk9VD8f+A6JbZfAxwUZT1vAFdWdiwgFdgJ9Cix7m7gs3KO2xDYCnSKvL8LeLKCOjYA+0de3w48V852Q4GcyOvDgO+BlBLrXwRuj7xeAVwMNCp1jDuA8UCXoP896KFHMj70OVZuHVF9jpWxX39gQ+R1G6AIaFrGdo8BDwT98w/zQy1htdfK4hdmlmpm90Samjez6y+dFuXsu845V1Di/TagQVkbmtlwM5sWuQy3ETiu1HHLO1ZL/I0jK0usW17eF+Ocy8X/tXhGZNEZlPgrOHKpYH7kMsVG/F+t5X195WkLrHTOFZWqqV3k9Sn4r2+5mX1iZgdHlt8LLAbei1zKuLGK5xWRsulzLIrPMTOrb2aPmdnyyPdmMtDEzFKBDsB659yGMnbtACyp7Piy9xTCwq+8y14ll58FjACOxv+nzoost+qc2MzSgdeA+4BWzrkmwMQoj7sWKMB/CBTrWMk+LwJnRsJPPeDjSB2HATcAI/F/7TUBNkVZR0mrgQ6l+nN1BFYBOOemO+dG4Jvz3wBeiSzPdc5d45zbF/glcLWZHVXFc4vUZvocq97n2DVAd+BA51wj/GVFIvuuBJqZWZMy9ltJBX3YpPoUwsLvB2DfSrZpCOwA1uH7Bdwdo3PXBdKJfBCZ2XBgWDQ7OucKgf8Dbo/8FdcLOK+S3Sbi+zzcAbxcosWqIf6DcC1Qx8xuxfftqKrP8ZcKrjezNDMbig9VL5lZXTM728waO+d2ApuBQgAzO8HMupiZlVheuBfnF6mt9DlWvc+xhvhLphvNrBlwW4kav8P3pXs00oE/zcyKQ9q/gAvM7CgzSzGzdmbWI8pzShQUwsLvz8AfzN+Zd2052zyDbyJfBcwDpsXixJGm9d/iW4Q24P9SnVCFQ1yOb9L/Ht+f46lKzrcD/4F3NL7TarF38R8yC/FfZx67Xx6IinMuHzgRGA78CDwKnOuc+yayyTnAskhz/xhgVGR5V+ADYAswFXjUOTepqucXqcX0OVa9z7EH8a1qP+K/L++UWn8Ovu/aN/j+bL+L1PIFkRsR8K1un+ADosSIRTrfiYiIiEgNUkuYiIiISAAUwkREREQCoBAmIiIiEgCFMBEREZEAKISJiIiIBCDpZp5v0aKFy8rKCroMEalBM2bM+NE51zLoOmJBn2EitUtFn19JF8KysrLIzs4OugwRqUFmVu5UL8lGn2EitUtFn1+6HCkiIiISAIUwERERkQAohImIiIgEIOn6hJVl586d5OTkkJeXF3QpSS8jI4P27duTlpYWdCkiIiKhFooQlpOTQ8OGDcnKysLMgi4naTnnWLduHTk5OXTu3DnockREREItFJcj8/LyaN68uQJYNZkZzZs3V4uiiIhIDQhFCAMUwGJE30cREZGaEZoQJiKyN8zsSTNbY2ZzyllvZvYPM1tsZl+Z2c9KrDvPzBZFHufVXNUiEgYKYTGwceNGHn300Srvd9xxx7Fx48Yq73f++efz6quvVnk/ESnT08CxFawfDnSNPEYDYwHMrBlwG3AgMBi4zcyaxrVSEQmVUHTMD1pxCLv00kt3W15YWEhqamq5+02cODHepYkknry1sPxl6H550JUA4JybbGZZFWwyAnjGOeeAaWbWxMzaAEOB951z6wHM7H18mHsxvhUnvh07YNMm2L7dv87P3/Uo+b6gAAoLdz2XfgTNzD9SUvZ8btwYjj/ev4+1tWvho49g505wDoqK9nyWYB1yCPTuXf3jKITFwI033siSJUvo378/aWlpNGjQgDZt2jBr1izmzZvHSSedxMqVK8nLy+PKK69k9OjRwK7pS7Zs2cLw4cM59NBDmTJlCu3atWP8+PHUq1ev0nN/+OGHXHvttRQUFDBo0CDGjh1Leno6N954IxMmTKBOnToMGzaM++67j//85z/88Y9/JDU1lcaNGzN58uR4f2tEdrdhFnwyAnasgbbDoeF+QVcUjXbAyhLvcyLLylu+BzMbjW9Fo2PHjvGpsgb9+CM88gjMn+/D1saNuz9qy709b78Nx1bUhlpF69bBfffBQw/B1q2xO67E3kMPKYSV6Xe/g1mzYnvM/v3hwQfLX3/PPfcwZ84cZs2axaRJkzj++OOZM2fOT8M8PPnkkzRr1ozt27czaNAgTjnlFJo3b77bMRYtWsSLL77IE088wciRI3nttdcYNWpUhXXl5eVx/vnn8+GHH9KtWzfOPfdcxo4dy7nnnsvrr7/ON998g5n9dMnzjjvu4N1336Vdu3Z7dRlUpFqWvwzTLoD05vCLz5IlgAGU1dbhKli+50LnHgceBxg4cGCZ2ySDH36Av/0NHn0Utm2DLl2gSRP/6NBh1+vGjf0jMxPq1t31SE/f/X2dOv6Rmlr2I+j7hMprhSoogCFD4NlnYxPCNmyA++/3v2e2boXTT/e/y5o127MFrvh10N+b2q5x49gcJ3QhLBEMHjx4t3G2/vGPf/D6668DsHLlShYtWrRHCOvcuTP9+/cH4IADDmDZsmWVnmfBggV07tyZbt26AXDeeefxyCOPcPnll5ORkcFFF13E8ccfzwknnADAkCFDOP/88xk5ciQnn3xyLL5UkcoVFcJXt8C8P0PLIXDoa1CvVdBVVUUO0KHE+/bA6sjyoaWWT6qxqmrQ99/DX/8K48b5y4lnnQU33ww9egRdWXBGjoR//xtyc6Fhw707xsaNPng98ABs3gynnQa33RabFhZJDqELYRW1WNWUzMzMn15PmjSJDz74gKlTp1K/fn2GDh1a5jhc6enpP71OTU1l+/btlZ7Hd1HZU506dfjiiy/48MMPeemll3j44Yf56KOPGDduHJ9//jlvvfUW/fv3Z9asWXuEQZGYyt8EU86C1ROhy2g44CFIrRt0VVU1AbjczF7Cd8Lf5Jz7zszeBe4u0Rl/GHBTUEXGw+rV8Je/wOOP+/5Jo0bB738Pkb/7arVRo3wofeMNOOecqu27Y4cPtfff74PYySf78NWvX3xqlcQVuhAWhIYNG5Kbm1vmuk2bNtG0aVPq16/PN998w7Rp02J23h49erBs2TIWL15Mly5dePbZZzniiCPYsmUL27Zt47jjjuOggw6iS5cuACxZsoQDDzyQAw88kDfffJOVK1cqhEn8bF4Ak0dA7hIYNBa6jgm6ojKZ2Yv4Fq0WZpaDv+MxDcA5Nw6YCBwHLAa2ARdE1q03szuB6ZFD3VHcST8M7rwT7rrLX3o77zy46SZ/+VG8Qw6BrCx/SbKqIewvf/Gh68QT4fbbYcCAeFQoyUAhLAaaN2/OkCFD6NOnD/Xq1aNVq12XWo499ljGjRtHv3796N69OwcddFDMzpuRkcFTTz3Faaed9lPH/DFjxrB+/XpGjBhBXl4ezjkeeOABAK677joWLVqEc46jjjqK/fffP2a1iOxm1USYciakpMNRH8I+hwddUbmcc2dWst4Bl5Wz7kngyXjUFaT//Q9uvRV+9SvfB0yzmO3JzLeG3X23bzFs2za6/fLy/E0Nxx0H48fHt0ZJfFbeJa1ENXDgQJednb3bsvnz59OzZ8+AKgoffT9lrzkH8+6B2TdD0/5w+BuQWf27Ac1shnNuYAwqDFxZn2GJpKgIBg/2/cAWLPCd66VsCxb4fnF/+xtcfXV0+zz9NFxwAbz/Phx9dFzLkwRR0eeXBmsVkdjYmQufngKzfw+dzvB3QMYggEnNeuYZmDED7rlHAawy3bvDoEHw3HPRbe+c74Tfpw8cdVR8a5PkoBCWwC677DL69++/2+Opp54KuiyRPW1eAO8OhlUT4GcPwCHPQ536QVclVZSb6/t+HXSQvwNSKjdqFHz5JcydW/m2H38MX33lh5/QEBMC6hOW0B555JGgSxCpXM54mHIOpGbAkR9Aq6FBVyR76e67/WXI8eP9eFRSudNP95cin3/ef/8q8uCD0LIlnH12zdQmiU//zURk7xQVwuxbYPJJ0KgHHDtDASyJLVnih0w491zfJ0yi06oVDBvmQ1hF0wktWgT//S9ccglkZNRcfZLYFMJEpOryN8Anv4S5f4L9LoRfTIbMDpXvJwnruusgLQ3+/OegK0k+o0bBihXw6aflb/P3v/vv7yWX1FxdkvgUwkSkajbMhncGwQ8fwKBxMPgJfylSktZHH8Hrr/uBWKMdakF2GTHC38RQXgf9DRvgqafgzDOhdeuarU0Sm0KYiETHOVg0Dt49EAq3w1GfQNeL1cM4yRUU+I7iWVnRD7Mgu8vM9KPe/+c/ZU9e/s9/+rk2f/e7mq9NEptCWAAaNGhQ7rply5bRp0+fGqxGJAo7N8P/zoDpl0Crn8PwWdDy4KCrkhh44gn4+mu47z71VaqOc86BTZvgrbd2X15QAA89BEOHQmR6YJGfxC2EmdmTZrbGzOaUs/5sM/sq8phiZhq+XSQRrf8S3v4ZrHwN+t8DQ9+CjJZBVyUxsGED3HKLDwgnnxx0NcntyCP9pcbSlyRfew1WroSrrgqmLkls8Ryi4mngYeCZctZ/CxzhnNtgZsOBx/GT41bPjN/BhlnVPsxumvaHA8qfGfyGG26gU6dOXHrppQDcfvvtmBmTJ09mw4YN7Ny5kz/96U+MGDGiSqfNy8vjkksuITs7mzp16nD//ffz85//nLlz53LBBReQn59PUVERr732Gm3btmXkyJHk5ORQWFjILbfcwumnn16tL1tqOedg0aMw82pIbwlHTYJ9Dg26KomhP/7RB7EHH9RV5epKTfVjqz30EKxfD82a+eUPPujn3DzhhGDrk8QUt5Yw59xkoNzJbJ1zU5xzGyJvpwHt41VLvJ1xxhm8/PLLP71/5ZVXuOCCC3j99deZOXMmH3/8Mddccw1VnSKqeJywr7/+mhdffJHzzjuPvLw8xo0bx5VXXsmsWbPIzs6mffv2vPPOO7Rt25bZs2czZ84cjj322Jh+jVLL5G+Cz0ZC9uXQ+mh/+VEBLFTmz/dzGP7mN6BpZGNj1CjYuRNeecW/nzbNP668UuOuSdkSZbDWC4G3Y3KkClqs4mXAgAGsWbOG1atXs3btWpo2bUqbNm246qqrmDx5MikpKaxatYoffviB1lW4Neazzz7jiiuuAKBHjx506tSJhQsXcvDBB3PXXXeRk5PDySefTNeuXenbty/XXnstN9xwAyeccAKHHXZYvL5cCbv1M3wA27oc+v8Vel4Dpt8gYXPTTb5D+Z13Bl1JePTvD716+UuSY8b4KYoaN4bzzw+6MklUgX+ymtnP8SHshgq2GW1m2WaWvXbt2porrgpOPfVUXn31VV5++WXOOOMMnn/+edauXcuMGTOYNWsWrVq1Iq+s22YqUF7L2VlnncWECROoV68exxxzDB999BHdunVjxowZ9O3bl5tuuok77rgjFl+W1CbOwaLH4L1DoCgfjp4Mva5TAAuhjRt9B/KLL/YjuEtsmPnWsP/9Dz75xPcH+81voIJ7saSWC/TT1cz6Af8ERjjn1pW3nXPucefcQOfcwJYJ+olxxhln8NJLL/Hqq69y6qmnsmnTJvbZZx/S0tL4+OOPWb58eZWPefjhh/P8888DsHDhQlasWEH37t1ZunQp++67L7/97W858cQT+eqrr1i9ejX169dn1KhRXHvttcycOTPWX6KE2c4tMPUcmD7G3/147JfQ8pCgq5I4efttf9feSScFXUn4FM+5edpp/jlyMUOkTIFdjjSzjsD/Aec45xYGVUes9O7dm9zcXNq1a0ebNm04++yz+eUvf8nAgQPp378/PXr0qPIxL730UsaMGUPfvn2pU6cOTz/9NOnp6bz88ss899xzpKWl0bp1a2699VamT5/OddddR0pKCmlpaYwdOzYOX6WE0qb58OkpsPkb6Hcn9P69Wr9Cbvx4P93OgdW/FUpK6dQJjjjCt4Sddhp07Bh0RZLIrKqdxaM+sNmLwFCgBfADcBuQBuCcG2dm/wROAYqbiAqccwMrO+7AgQNddnb2bsvmz59Pz549Y1d8LafvZy2y7AX4YjTUyYRDXoDWRwVdUZnMbEY0nw/JoKzPsJqUn+8vQY4c6ccIk9h7+mm44AKYOhUOOijoaiRoFX1+xa0lzDl3ZiXrLwIuitf5RaQChXkw4ypYPA5aHgpDXob6mq+mNpg0CTZv9lPtSHycdx4cdhjst1/QlUiiS5S7I2udr7/+mnPOOWe3Zenp6Xz++ecBVSS1xuZFMOVMfxdkz+tg/7sgJS3oqqSGjB8P9evDUYnZ6BkKZgpgEh2FsID07duXWbNiPKisSEWcgyX/9AMap9SFw9+A9moOqU2cgwkT4JhjoF69oKsRkdD0vo1X37baRt/HkMpbA5NP8v2/WhwMx3+tAFYLzZwJOTm6FCmSKELREpaRkcG6deto3rw5prk39ppzjnXr1pGhWXzDZdVb8Pmv/Sj4P7sful+pux9rqfHj/cjtxx8fdCUiAiEJYe3btycnJ4dEHcg1mWRkZNC+fdLOICUlFWyDL6+FRWOhSV848gP/LLXW+PFw6KHQokXQlYgIhCSEpaWl0blz56DLEEkc67JhytmQuwh6Xgv9/gSp6UFXJQH69lv46iu4776gKxGRYqEIYSIS4Ypg/r0w+w9QrzUc9aEfAV9qvQkT/LP6g4kkDoUwkbDY/j1MPRe+fx86ngaDH4O6TYOuShLE+PF+cukuXYKuRESKqXeuSBisfhfe3h/WfgaDH/eDryqAScT69TB5slrBRBKNWsJEkllhPnx1M8y/Dxr3gaM+hsa9gq5KEszEiVBYqBAmkmgUwiQYS/8N27+D3jcGXUnyyl0C/zsT1k+HrpfAgL9BHY3AKXsaPx7atIFBg4KuRERKUgiTmrd1BUy/BNIaKoTtrWUvwBdjwFLhsNegw8lBVyQJascOeOcdOOssP0aYiCQOhTCpeV9eD4Xb/aNgu1pvqqJgO8y4EpY8AS2HwCHPQ2anoKuSBPbRR7Bliy5FiiQi/V0kNeuHT2DFy77/EsDW5cHWk0w2L4L3DvYBrNdNcNQkBTCp1PjxkJkJRx4ZdCUiUppCmNScokLfilO/g58+BxTCorXiVXjnANi2Ao74L/S/G1LUkC0VKyry44MdeyxoNjKRxKNPcak5S56AjbPh0FegcU+/bOuyQEtKeIX58OV1sPAf0PxAOPRltX5J1LKz4bvvdClSJFEphEnNyN8AX/0B9jkCOpzqR3a3OmoJq8jW5fDZSFj3BXT/HfT/C6TWDboqSSLjx0NqqibsFklUCmFSM766zQexA/4BZv6uvsyOagkrz6r/+tHvXSEc+ip0PCXoiiQJjR8Phx0GzZoFXYmIlEV9wiT+Ns6BRY9Cl4uhab9dyzM7qSWstO0/wJRz4JNf+u/PsTMUwGSvLFkCc+fqUqRIIlMIk/hyznfGT2sE/e7cfV1mllrCihUVwsJH4b/d/d2jvW+GX0yBhproT/bO+PH+WSFMJHHpcqTEV87r8MNHMPBhSG+++7rMTn7U/MIdkJoeTH2JYF22H7x2fTa0OhIGPQqNugddlSS5iROhTx/o3DnoSkSkPGoJk/gp2A4zr4Emff2lyNIyswAH21bWdGWJIX8jTL8M3h0M23LgkBfgyA8UwKTanINZs+Cgg4KuREQqohAm8fPN3/zlxgP+XvaYVsVDLdS2fmHO+WmH/tsDFo+DbpfDCd9A1pn+pgWpUWZ2rJktMLPFZrbHPFpm1snMPjSzr8xskpm1L7Gu0MxmRR4Tarby8q1ZA+vWQe/eQVciIhXR5UiJj60rYe7dfjiKVj8ve5vMrMi2y2qqquBt/w6+uBhWvQnNBsHQt6DZAUFXVWuZWSrwCPALIAeYbmYTnHPzSmx2H/CMc+7fZnYk8GfgnMi67c65/jVadBTmzvXPffoEW4eIVEwtYRJ7O9bB/04HHPzsvvK3q98OLKV2tIQVt3691Ru+fx8G/A2GTVUAC95gYLFzbqlzLh94CSjdlb0X8GHk9cdlrE84c+b4Z7WEiSQ2hTCJrS3L4P0hsH4mHPxcxaO7p6RBvfZ+nzDLWwOfngJTzoaG3WH4LOh5NaSkBl2ZQDugZKfEnMiykmYDxeOE/ApoaGbFd5lkmFm2mU0zs5PKO4mZjY5sl7127dpY1V6uuXP92GCtW8f9VCJSDbocKbGzYRZ8PBwK8+DI92CfwyvfJ7MTbAtxS9iK/8D0S2FnLvT/K/RQ+EowZXXCc6XeXws8bGbnA5OBVUBBZF1H59xqM9sX+MjMvnbOLdnjgM49DjwOMHDgwNLHj7k5c3wrmLoYiiQ2tYRJbHz/Ibx/uO+A/4vPogtg4PuFhbElLO9H+Ox0P+1QZmcYPhN6XacAlnhygA4l3rcHVpfcwDm32jl3snNuAHBzZNmm4nWR56XAJGBADdRcIed8S5j6g4kkPoUwqb5vn4dJw32r1rCp0KQKHVEyO8H2VVBUUPm2ycA5+PY5eKuXHyNt/7th2BRo3CvoyqRs04GuZtbZzOoCZwC73eVoZi3MrPiz8ibgycjypmaWXrwNMAQo2aE/EKtWwaZN6g8mkgziFsLM7EkzW2Nmc8pZb2b2j8ht4V+Z2c/iVYvEiXMw768wdRS0GAK/+BTqt698v5IaZPn5EbflxKXEGrXpG/joKJh6DjTYz0851PumsofnkITgnCsALgfeBeYDrzjn5prZHWZ2YmSzocACM1sItALuiizvCWSb2Wx8h/17St1VGQjdGSmSPOL52+Fp4GHgmXLWDwe6Rh4HAmMjz5IMigph5lWw8CHoOBIOfmbvRr0vOVZYg6yYllhjCrb74Tjm/wVSM2HwY7DfRf7OT0l4zrmJwMRSy24t8fpV4NUy9psC9I17gVWkOyNFkkfcQphzbrKZZVWwyQj82DsOmGZmTcysjXPuu3jVJDFSsN239qx8Dbpf5Yeh2NvAsdtYYUfEqMAatPpdyL4MtiyBrHNgwL1Qr1XQVUktNncutGoFLVoEXYmIVCbI6yTl3RquEJbIdqyDT06EH6f4sa56Xl2949WP9IlOtrHCtq2GmVf7ybYbdoMjP4TWRwZdlchPd0aKSOILMoRFc2u439BsNDAaoGPHjvGsSSqyZakfgmLrcjj0Feh4WvWPmZoO9domz6j5rggWjYPZN/mJx/veAb2ur90TkEvCKCqCefPgwguDrkREohFkCKv01vBiNT3GjpRh3XT45AQo2uknmd7n0NgdO7NTcrSEbfwaPh8N66ZBq6Ng0Fho1DXoqkR+snw5bN2qljCRZBFkz+EJwLmRuyQPAjapP1iCWvVf+GAopNaHX0yJbQAD3y8skVvCCrbBrBvh7Z/BlsVw8LNw5PsKYJJwdGekSHKJW0uYmb2Iv7W7hZnlALcBaQDOuXH4u5GOAxYD24AL4lWLVMOicb7jedMBcMR/oV4c5kHJ7AQrX/V3XCbaYKbfvQfTL/GXYve9wHe8T29e+X4iASi+M7KXhqUTSQrxvDvyzErWO+CyeJ1fqskVweybYd490PY4GPIypDWIz7kys/xlzrzvqj7OWLzkrYEZV8HyF3zH+6M+hlZDg65KpEJz50L79tCkSdCViEg0NIqk7KkwHz6/EJY9B11Gw8BH4jvgaMmxwoIOYUWFsPgxH0ALt0Gf26D3jZCaEWxdIlHQnZEiyUUhTHa3Mxc+PQW+fx/63Qm9b47/LMDFY4VtWQYth8T3XBVZO9Vfet3wJbQ6EgY+DI17BlePSBUUFsL8+XDUUUFXIiLRUgiTXbb/AJOOg42z4cB/wX6/rpnzZkaGHdkW0B2SeWtg1g2w9Gmo185feu14WvzDp0gMLVkCO3aoU75IMlEIEy93MXx8DGz/Dg4fD+2Or7lz16kPGfv4lrCaVFQAi8bCV7f4S4+9boDef4hf3zeROCq+M1KXI0WSh0KYwLps3wJGke+A3iKAKTzr1/BYYT9Ogy/G+Fa/1kfDAQ9B4x41d36RGNOdkSLJRyGstlv9Dnx2KqS3hJ+/A426B1NHgyzYMLtmzrXoMci+HOq1gUP/Ax1O0aVHSXpz50LnzpCZGXQlIhKtIAdrlaB9+yx88kto2BWGTQkugIG/Q3LbCnBxnB2DXMUAACAASURBVBChqACyr4DpY6DNMDjua+h4qgKYhMKcOeoPJpJsFMJqq/n3wdRzYZ8j4OhPfKtQkDKzoDAP8n6Iz/HzN8Ck4bDwYehxDRw+Aeo2js+5RGpYfj4sWKD+YCLJRpcjaxvnIoOw/hk6joSDn0mMyadLjhUW61H5Ny/wLX5bl8GBT8J+mpxBwmXRIigoUEuYSLJRS1htUlTop+CZ92focjEc8kJiBDDYNVZYrOeQ/O59ePcgyN/obzpQAJMQ0p2RIslJIay2KMyHKWf70eB73QiDxibWPI0lW8JiwTlY8JC/BJnZAY75ItiBYEXiaM4cSEmBHrrBVySp6HJkbVCwDT49Fb57G/r/BXpdH3RFe0prCHWbxaYlzDmY8Vvf/6vdiXDIc/74IiE1dy506QIZml1LJKkohIVd/kb45ARYOwUGPw5dfhN0ReXLjMFYYc7BzGsiHfCvhgH3gqnBV8JNd0aKJCf9dgqz7T/AB0Nh3Rdw6MuJHcDA9wurbkvYnDtgwQPQ7bcw4D4FMAm9vDxYvFj9wUSSkX5DhdXWlfDBYZC7CA5/08+FmOiKW8L2dqyw+ffD17fDvhfAAQ9o/C+pFb75BoqK1BImkowUwsKoYDtMPgm2fw9Hvg9tjwm6ouhkZkHBVtixrur7Ln4CvrzGh83BT6gFTGoN3Rkpkrz0mypsnIPsS2HDTN8hveUhQVcUveI7JLdVsV/Yspfgi4uhzXA4+LnEuutTJM7mzIG0NOjaNehKRKSqFMLCZvFjsPRp6HMLtD8x6GqqpkGWf96yLPp9ct6EqefAPofDYa9Bat14VCaSsObOhW7doK7+6YskHYWwMFk71Q/N0GY49Lkt6GqqrqpjhX3/EXx2GjQdAEdMgDr14lebSILSnZEiyUshLCy2/wCfnQr12vvLkMl4SS6tCaQ1iu4OyR+nweQT/eTjP3/b7ydSy2zdCt9+qxAmkqw0TlgYFO2E/430k1QPmwrpzYKuaO+YRTdWWFEBTBkFGa3gyPcgvXnN1CeSYObN88/qlC+SnBTCwuDL62HNZDj4WWi6f9DVVE80Y4Wt/D/YsgQO+z+o16YmqhJJSMV3RqolTCQ56XJkslv2Iix40A9O2nlU0NVUX2UtYc7BvL9Aw25+SiKRWmzOHD9V0b77Bl2JiOwNhbBktuEr+PxCaHkY/Oy+oKuJjcws2LnJT7dUlh8+8sNv9LwuOfu9icTQ3LnQsyek6r+CSFJSCEtW+Rvh05OhbhM49BVISQu6otj46Q7JZWWvn/cXyGgdjlY/kWqaM0f9wUSSmUJYsvrqVh9UDn0V6rUOuprYyczyz2Vdklw/E75/H3r8DlIzarQskUSzaRPk5Kg/mEgyUwhLRpsXwaKxsN9FyTUifjSKW8LKGrB1/r1QpyF0GVOjJYkkIk1XJJL8FMKS0ewbfUtQ39uDriT20ltAav09W8K2LIUVr0DXMVC3cTC1iSQQ3RkpkvwUwpLN2v/5IRp6Xh+uy5DFfhorbNnuy+ffD5YK3X8XSFkiiWbOHMjMhI4dg65ERPZWXEOYmR1rZgvMbLGZ3VjG+o5m9rGZfWlmX5nZcfGsJ+k5BzOvhXptoefVQVcTP5lZu7eE5a2FpU9C1jlQv21gZYkkkkWL/JyRKfpTWiRpxe2/r5mlAo8Aw4FewJlm1qvUZn8AXnHODQDOAB6NVz2hsPJVWDcN+t0JdTKDriZ+SreELXwICrf7YSlEBPAd85s2DboKEamOeP4NNRhY7Jxb6pzLB14CRpTaxgHFk/41BlbHsZ7kVrgDZt0ITfpC5/OCria+GmRB/nrYmQs7t8DCh6H9CGjcI+jKRBJGbi40bBh0FSJSHfGctqgdsLLE+xzgwFLb3A68Z2ZXAJnA0XGsJ7ktGus7pw99J/yDlNYvHitsOXz/oZ8Ts+cNwdYkkmAUwkSSXzxbwqyMZa7U+zOBp51z7YHjgGfNbI+azGy0mWWbWfbatWvjUGqCy98Ac+6A1sOg7TFBVxN/DbL8c+5i+OZ+PyNAy4MDLUnCK4q+q53M7MNIv9VJZta+xLrzzGxR5FGjTdQKYSLJL54hLAfoUOJ9e/a83Hgh8AqAc24qkAG0KH0g59zjzrmBzrmBLVu2jFO5CWzu3X6E/AH3Bl1JzSgeK2z+X2HbCuh1fbD1SGhF2Xf1PuAZ51w/4A7gz5F9mwG34Vv4BwO3mVmN9dJSCBNJfvEMYdOBrmbW2czq4jveTyi1zQrgKAAz64kPYbWwqasCW76FBf+Afc+Hpv2CrqZmZLSClHT4cSo07g1tddOsxE00fVd7AR9GXn9cYv0xwPvOufXOuQ3A+8CxNVAz+fn+oRAmktziFsKccwXA5cC7wHz8XZBzzewOMzsxstk1wG/MbDbwInC+c670JcvabfbNfnysfncGXUnNsRTIjAx+1PM6/14kPsrqu9qu1DazgVMir38FNDSz5lHuGxe5uf5ZIUwkucWzYz7OuYnAxFLLbi3xeh4wJJ41JLV102H5i9D7D1C/Rj7bE0fDblCYB53ODLoSCbdo+q5eCzxsZucDk4FVQEGU+/qTmI0GRgN0jMHoqgphIuEQ1xAm1eAcfHktZOxTO/tEDX4civIhtW7QlUi4Vdp31Tm3GjgZwMwaAKc45zaZWQ4wtNS+k8o6iXPuceBxgIEDB1a7tb84hDVqVPF2IpLYdJ0nUS17AdZM9vNDptXCP3frt911l6RI/FTad9XMWpS4a/sm4MnI63eBYWbWNNIhf1hkWdypJUwkHBTCEtGmb2D6xdDyUNjvN0FXIxJaUfZdHQosMLOFQCvgrsi+64E78UFuOnBHZFncKYSJhIMuRyaagm3w2WmQWg+GvAQp+hGJxFMUfVdfBV4tZ98n2dUyVmMUwkTCQb/hE0325bBpLvz8ndrXGV9EoqIQJhIOuhyZSJY8BUufgj63QJthQVcjIglKIUwkHBTCEsXGryH7Mmh1JPS5tfLtRaTWUggTCQeFsESwM9f3A0trDIe8EP4JukWkWnJzoW5d/xCR5KU+YUFzDr4YDbmL4MiPoF6roCsSkQSneSNFwkEhLGiLH4PlL8H+d0OrI4KuRkSSgEKYSDjocmSQ1s+EGVdCm+HQ64agqxGRJLF5s0KYSBgohAVl52bfDyyjFRzyrCapFpGoqSVMJBx0OTIoix+HLUvh6E8hvXnQ1YhIEsnNhWbNgq5CRKpLzS9BcEWwaJyflmifQ4OuRkSSjFrCRMJBISwI370PW5ZA10uDrkREkpBCmEg4KIQFYfFYSG8JHU4OuhIRSUIKYSLhoBBW07augFVvwn4XQWp60NWISJJxDrZsUQgTCYOoQpiZXWlmjcz7l5nNNDNNbrg3Fj/hP0W7Xhx0JSKShLZtg6IihTCRMIi2JezXzrnNwDCgJXABcE/cqgqrwnxY8gS0PR4yOwVdjUiomNmvzKxxifdNzOykIGuKB80bKRIe0YYwizwfBzzlnJtdYplEK+d1yPsBuqlDvkgc3Oac21T8xjm3EbgtwHriQiFMJDyiDWEzzOw9fAh718waAkXxKyukFo2FzM7Q5pigKxEJo7I+z0I3FqJCmEh4RPsBdSHQH1jqnNtmZs3wlyQlWhvnwppPoP9fNDq+SHxkm9n9wCOAA64AZgRbUuwphImER7Rp4GBggXNuo5mNAv4AbKpkHylp8ThIqQv7KruKxMkVQD7wMvAKsB24LNCK4kAhTCQ8om0JGwvsb2b7A9cD/wKeAY6IV2GhsnMLLP03dBwJGS2DrkYklJxzW4Ebg64j3hTCRMIj2pawAuecA0YAf3fO/R3QR0C0lr8ABbkaIV8kjszsfTNrUuJ9UzN7N8ia4kEhTCQ8om0JyzWzm4BzgMPMLBVIi19ZIeIcLHwUmuwPLQ4KuhqRMGsRuSMSAOfcBjPbJ8iC4qE4hDVqFGwdIlJ90baEnQ7swI8X9j3QDrg3blWFyY/TYONsPyyFaVQPkTgqMrOOxW/MLAvfQT9UikNYg62TIGdCoLWISPVE1RLmnPvezJ4HBpnZCcAXzrln4ltaSCx6FOo0hE5nBV2JSNjdDHxmZp9E3h8OjA6wnrjIzYX9Oy8kZfLxULgdDh8P7X8ZdFkisheinbZoJPAFcBowEvjczE6NZ2GhkLcWVrwC+54HaQ2CrkYk1Jxz7wADgQX4OySvwd8hGSrbtuzk6dFnQ2oGNO0PU86EDbODLktE9kK0lyNvBgY5585zzp0LDAZuqWwnMzvWzBaY2WIzK/OuJTMbaWbzzGyumb0QfelJYOlTUJQPXS8JuhKR0DOzi4AP8eHrGuBZ4PYga4qHYW1vp3/HbDjwn3DEfyGtCXzyS9j+fdCliUgVRRvCUpxza0q8X1fZvpHO+48Aw4FewJlm1qvUNl2Bm4AhzrnewO+iLTzhuSJYNA72OQIa96p8exGpriuBQcBy59zPgQHA2mBLirE1kxnR5c+88fWF0OFXUL8tHPEm7FgHk0dAQega/kRCLdoQ9o6ZvWtm55vZ+cBbwMRK9hkMLHbOLXXO5QMv4Ye4KOk3wCPOuQ0ApYJecvthEmz9FrqMCboSkdoizzmXB2Bm6c65b4DuAdcUO/kbYcooVm/ej8emP7hrebMBcMjzsG46TDvf/wEoIkkh2o7515nZKcAQ/MTdjzvnXq9kt3bAyhLvc4ADS23TDcDM/gekArdH+nUkv5zXIbUetD8x6EpEaoucyDhhbwDvm9kGYHXANcWGc/DFGNj+HTe+OYU69Ur1Me1wkp8Sbdb10KgH9PtjMHWKSJVEPbmtc+414LUqHLus8RhK3y5eB+gKDAXaA5+aWZ+SY/0AmNloInc5dezYkYTnHOS84SfqrlM/6GpEagXn3K8iL283s4+BxkA4/qhb9hyseBn2v4tpCwcxeHAZ2/S8FjZ/A3PugIbdoPPZ8amlcAdsWwlbl/u7MyuS0RoadoW6jeNTi0iSqzCEmVkuZY+zY4BzzlU0XGAO0KHE+/bs+VdpDjDNObcT+NbMFuBD2fSSGznnHgceBxg4cGDij/uzPhu25UC/u4KuRKRWcs59UvlWSWLLtzD9Mmh5GPS8gdzcckbLN4NBY2HLEvj819CgM7Q8ZNd6VwSbF8CPUyOPaVCwFdKbQd3mkB55lHxdsAW2LIOty3zo2roMtn9HlYdfy2jlg2HDrtCo267XGftAnQb+qkG04ygWFfgZSAq2+eOmRN2WUHs553+WO9ZB/jr/7BxkdoLMjmosCFCF/3qdc9WZGGM60NXMOgOrgDOA0oNlvQGcCTxtZi3wlyeXVuOciWHl62Cp0O6EoCsRkWRWVABTRoGlwCHPQkpq+SEMILUuHPYavHsQTD7Jh7JNcyKh63PYGbnIULcpND/IB7AdkV/KWxb7552bdj+m1YH6Hfwv7DbDIDMr8su7kx8DsTyuCLavhtyF/rF5Iax+C5Y+uee2lurDWFpD/1ynoR/Wp3CHD1w7t0SC1xYozNu1X0pdaNQTmvSBxn38c5O+UL/jrlBXVOjrKA6RW5f7x/bv/Fy+P309WdAgC+q1S8xgV7jD/yzXz4T1M3w4rygMF+3Y9bPNX+/v1C9Pxj5Qv5P/+ou/H2mN/fd7Z26J5xKvsV1hvWR4r9vMP2O7At9P4W99iRBYGPk5N9z18/7pfQOo22T3PwjSmkBKaky/pYkgbv/SnHMFZnY58C6+v9eTzrm5ZnYHkO2cmxBZN8zM5gGFwHXOuXXxqqnG5Lzh74pMbxZ0JSKSzObeDT9OgUNegMxOFBTA9u2VzBuZ3tzfMfnewfDZqYBB497Q8TRocbB/NOrmg11Zigr8L+0d6/wvw3ptY/vLL38T5C7yj/z15f+SL9jiW8iKW8uKf1kXv07N8EFk49ewZjIse37XOeo09F9j/gbYugJcwe41ZOwDGW1gw4xIy14Jlgr120OD/aDjqX6g7VhfTl01ERY9AnUyy2+FBNgwyweu9TN8ACva6ZenNYn8DCv4uaSk+RbHFmUcu27k+FuXw7blkdbO5X68uZwJPsCVVvpnQBFs/MqHqoKt0X3dqfVLfH0pULA4ErIjP+8Kmf/jobj+1IwKtnX+Z16Y77+Wovxdj8LI+8pac1Prlf3vrjgkdjgV9jk0uq+7AnGN+865iZS6i9I5d2uJ1w64OvIIh80LYPN8TdYtItWzdqrv35U1CrLOBGBL5PdUpZN3N+4Bw6bC9hxoNqhqISKlTiSkxGnazbqNoflA/4il/E0+qGyc4583L/QhpOPI3Vu7Sl9+K8yDrSt3DyNbl8GGL2H6pTDzWug0Evb7jQ+w1Zl+rqgAvr7Nh+v6kTqKW4nKu6u1bjNodgD0uNo/NzsAMjvHaBq8MkKEK4K8NbBzc4kAUr/80A7+e1iylWvHOsDtGf4qCk6uyF9iLg7h+RvKaEkr8bqilj2AlAyo08i3DqfUhZT0yHPkUdHXg/P9HUv+gbBjDWxd6l/vzPVDTyV6CKuVVkZuGm1fejQOEZEqmPsnfxlw4MM/LSqeN7LSEAY+iDXuEZ/aElHdxtByiH9URWoGNOrqHyU55/v3Ln4Clr8IS5/2LYr7XQSdz9nVWhWt7d/DlLPgh499oDvg71CnXuRcRf4y8I4SQcMVQJN+PjzW5LzDlgL1WvtHtFIz/Jh19dtW77xpDfyjKudOcgphsZbzBjQbCJkdKt9WRKQ8h/7HX0or0YpVpRAm1WMGzQf5x8/+BstfhiVPwMyrYNaN0OEU2Pd8aHVk5Zdr10yG/53hx3o76Gk/ld1u50rxl9rqNoWGXeL1FUkCinawVonGtlWw7nM/krWISHXUqb9HS5ZCWEDSGkKXi+CYz2H4LN8atnoifDwMxneCWTfBpm/23M85mPdX+PBI34/omM/3DGBSqymExVLOeP/cXiFMRGJPISwBNN0fBj0MJ38HQ1727+ffC2/1hHcPhIWPRPotbfR3qM66wf9OODbb37kpUoIuR8ZSzuvQqDs07hl0JSISQgphCSQ1w3fY7zTS9/da9gJ8+2/Ivtxfsqzb1HdWP+Dv0O2Kmu3XJUlDISxW8jf4+SJ7XhN0JSISUgphCapea+h5tX9smAVL/+2Hzuh3J7Q8OOjqJIEphMXKqrf83Sy6FCmSVMzsWODv+PEM/+mcu6fU+o7Av4EmkW1udM5NNLMsYD6wILLpNOfcmHjWqhCWBJr2hwP6B12FJAmFsFjJed0Path8UNCViEiUzCwVeAT4BX4atelmNsE5N6/EZn8AXnHOjTWzXvixD7Mi65Y452rsN65CmEi4qGN+LBRsh9XvQPuTKhkATkQSzGBgsXNuqXMuH3gJKD3InwOK58ltzJ5z4NaY3FxISYF69YKqQERiSYkhFr5/Hwq3+RAmIsmkHbCyxPucyLKSbgdGmVkOvhXsihLrOpvZl2b2iZkdFtdK4ad5I9XHWyQcFMJiIed1P5dXq6FBVyIiVVNWnCk9qdyZwNPOufbAccCzZpYCfAd0dM4NwE+99oKZNaIMZjbazLLNLHvt2rV7XWyFk3eLSNJRCKuuogJY9Sa0O8FPmCoiySQHKDm9RXv2vNx4IfAKgHNuKpABtHDO7XDOrYssnwEsAbqVdRLn3OPOuYHOuYEtW7bc62Jzc6FRmTFPRJKRQlh1rf3MD8ynUfJFktF0oKuZdTazusAZwIRS26wAjgIws574ELbWzFpGOvZjZvsCXYGl8SxWLWEi4aK7I6tr5et+0L42xwRdiYhUkXOuwMwuB97FDz/xpHNurpndAWQ75yYA1wBPmNlV+EuV5zvnnJkdDtxhZgVAITDGObc+nvUqhImEi0JYdTjnJ+xuPQzqZAZdjYjsBefcRHyH+5LLbi3xeh4wpIz9XgNei3uBJeTmQuvWNXlGEYknXY6sjg0zYdsKXYoUkRqhljCRcFEIq46Vb/hxwdr9MuhKRKQWUAgTCReFsOrIeR1aHg7pzYOuRERCzjmFMJGwUQjbWxvnwqa5uhQpIjVixw7YuVMhTCRMQhvCcnPh8svhnXfidIJv/gap9aDTWXE6gYjILpo3UiR8QhvC6teHJ5+E996Lw8G3rYJlz8F+F0JGizicQERkdwphIuET2hCWmgq9e8PXX8fh4AseBFcEPa6Jw8FFRPakECYSPqENYQD9+sFXX8X4oPkbYdFj0HEkNMiK8cFFRMqmECYSPqEOYX37wpo18MMPMTzo4segIBd6XhfDg4qIVEwhTCR8Qh3C+vXzzzFrDSvMg28e9CPkNxsQo4OKiFROIUwkfEIdwvr29c8x6xf27XOQ9z30uj5GBxQRiY5CmEj4hDqEtWwJbdrEqCWsqBDm3wvNDoBWR8bggCIi0VMIEwmf0E/g3bdvjELYqgmQuxCGvAxmMTigiEj0FMJEwifULWHg+4XNmwcFBdU4iHMw7y/QYF/ocHLMahMRiVZuLqSnQ1pa0JWISKzENYSZ2bFmtsDMFpvZjRVsd6qZOTMbGOsa+vXz030sWlSNg6z9FNZ9Dj2vhZTQNx6KSALSvJEi4RO3EGZmqcAjwHCgF3CmmfUqY7uGwG+Bz+NRR0zukJz3V0hvCZ3Pj0VJIiJVphAmEj7xbAkbDCx2zi11zuUDLwEjytjuTuCvQF48iujRw4+ev9chbOMcWP0WdLsC6tSLaW0iItFSCBMJn3iGsHbAyhLvcyLLfmJmA4AOzrn/xquI9HQfxPY6hM2/F1LrQ7fLYlqXiEhVKISJhE88Q1hZtxC6n1aapQAPAJVOwGhmo80s28yy165dW+VC+vXby7HCtq6EZS9Al99AerO9OICISGwohImETzxDWA7QocT79sDqEu8bAn2ASWa2DDgImFBW53zn3OPOuYHOuYEtW7asciF9+8Ly5bBpUxV3/OYBwEGPq6p8ThGRWFIIEwmfeIaw6UBXM+tsZnWBM4AJxSudc5uccy2cc1nOuSxgGnCicy471oUUd86vUmtY3lpY8jh0OhMyO8W6JBGRKlEIEwmfuIUw51wBcDnwLjAfeMU5N9fM7jCzE+N13rLsVQibfRMU7oDev49LTSIiVZGbC40aBV2FiMRSXAe9cs5NBCaWWnZrOdsOjVcd7dtDkyZV6Jz/4zRY8i/oeR007hmvskREouIcbNmiljCRsAn9iPngZxmKevqiokKYfhnUawt9bol7bSIildm61QcxhTCRcKkVIQx23SHpXCUbLnkCNsyEAX+DNH3iiUjwNG+kSDjVqhCWm+vvkixX3o8w+/fQ6ufQ6fQaq01EpCIKYSLhVGtCWN++/rnCS5Kzb4KduTDwYX8NU0QkASiEiYRTrQlhffr453JD2I+f+8743a+ExntMcSkiEpjNm/2zQphIuNSaENawIey7bzkhrKgQsi+Deq2h7201XpuISEXUEiYSTnEdoiLRlDt90ZJ/wvoZcMgL6owvIglHIUwknMLdEvbN32H9lz/dEtm3LyxcCNu3l9imuDP+PkdApzOCqVNEpAIKYSLhFN6WsLw1MOt6KMr3fbyyzuagvmdRVJTFvHlwwAGR7Wb/HnZuUmd8EUlYCmEi4RTelrCMfeBX38GgcVC3Ocy+meN2dubTWw9lx5yxsGMd/PiFvxTZ/Upo0ifoikVEylQcwjIzg61DRGIrvC1hAOnNoOvF/rFlGUXfvkizVc/RK+1S+L/fQt2mkNFKnfFFJKHl5kKDBpAS3j+bRWql2vNfukEWKX1v4rxX5nDxG19Cj99BenMYNBbSNCuuiCSu3FxdihQJo9oTwiL69TNen9QfBtwLJ8yHDicFXZKISIUUwkTCqRaGMFi7Fn74IehKRESioxAmEk61LoRFNX2RiEgCUQgTCSeFMBGRBKcQJhJOtS6EtWwJbdoohIlI8lAIEwmnWhfCoILpi0Sk1jGzY81sgZktNrMby1jf0cw+NrMvzewrMzuuxLqbIvstMLNj4lWjQphIONXKENa3L8ydCwUFQVciIkEys1TgEWA40As408x6ldrsD8ArzrkBwBnAo5F9e0Xe9waOBR6NHC/mFMJEwqlWhrB+/SA/388jKSK12mBgsXNuqXMuH3gJGFFqGwcUDybYGFgdeT0CeMk5t8M59y2wOHK8mCoogLw8hTCRMKq1IQzUL0xEaAesLPE+J7KspNuBUWaWA0wErqjCvtWmeSNFwqtWhrAePaBOHfULExGsjGWu1Pszgaedc+2B44BnzSwlyn39ScxGm1m2mWWvXbu2SgUqhImEV60MYenp0L27WsJEhBygQ4n37dl1ubHYhcArAM65qUAG0CLKfYns97hzbqBzbmDLli2rVKBCmEh41coQBv6SpEKYSK03HehqZp3NrC6+o/2EUtusAI4CMLOe+BC2NrLdGWaWbmadga7AF7EusDiENdIUtyKhU6tD2IoVsGlT0JWISFCccwXA5cC7wHz8XZBzzewOMzsxstk1wG/MbDbwInC+8+biW8jmAe8AlznnCmNdo1rCRMKrTtAFBKV45Pyvv4ZDDw22FhEJjnNuIr7Dfcllt5Z4PQ8YUs6+dwF3xbM+hTCR8Kq1LWH77++fP/882DpERCqiECYSXrU2hLVvD4MGwZNPgivzfiYRkeBt3uyfFcJEwqfWhjCAMWNg3jz47LOgKxERKZtawkTCq1aHsNNPh8aNYdy4oCsRESlbbq4f1zA9PehKRCTW4hrCopgY92ozmxeZFPdDM+sUz3pKy8yEc8+FV1+FKo6fKCJSI4rnjbSyhoYVkaQWtxAW5cS4XwIDnXP9gFeBv8arnvKMGePnkXzqqZo+s4hI5TR5t0h4xbMlrNKJcZ1zHzvntkXeTsOPOF2jevWCww+Hxx6DoqKaPruISMUUwkTCK54hrKqT214IvF3WiurMuxaNMWNg6VL44IOYH1pEpFoUwkTCK54hrCqT244CBgL3lrW+OvOuRePkk6FlS3XQF5HEoxAmEl7xDGFRTW5rZkcDNwMnOud2xLGecqWnw69/DRMmwKpVQVQgIlI2hTCRDHbBuwAAFTRJREFU8IpnCKt0YlwzGwA8hg9ga+JYS6VGj4bCQvjnP4OsQkRkdwphIuEVtxAW5cS49wINgP+Y2Swzm1DO4eJu333hmGPgiSegoCCoKkREdqcQJhJecZ3AO4qJcY+O5/mraswY+NWv4K3/b+/eg6sqzz2Of5+EQLjJJYjNEAVEq5iCQRhvcOTMybGKLQUZbGPVIlOltFgErULFC0elpYq2eGkUD6ihtEKxFHtKbzZRphXlZiIGbVG0Q0QBwUCiRgO85493hYawE3LbWSt7/z4za/ba67af/ZL98qx3rfW+v4dx446/vYhIPDmnJEwkkSV1j/l1ffWr0K+fbtAXkWioqvK3SSgJE0lMSsJq6dABrr8e/vQn32WFiEiYNG6kSGJTElbHdddBSgosWhR2JCKS7JSEiSQ2JWF19OsHY8fCkiXwWSgdZoiIeErCRBKbkrAYpk71A3qvWhV2JCKSzJSEiSS2uD4d2V5dfLHvsuKxxyAvL+xoRCRZKQmTtlRdXU1ZWRlVVVVhh9Iupaenk5WVRVpaWqP3URIWQ0oKfOc7MGsWbN3qB/kWEWlrSsKkLZWVldG9e3cGDBiAWayRB6U+zjn27t1LWVkZAwcObPR+uhxZj8mToWtXuO22sCMRkWSlJEzaUlVVFRkZGUrAmsHMyMjIaHIropKwepx4Itx+O6xeDX/4Q9jRiEgyUhImbU0JWPM1p+yUhDVg5kw4/XS48UY9KSkibU9JmEhiUxLWgE6d4OGHYds2+OlPw45GRJJNRQV07uw7khZJdOXl5fz85z9v8n6XXXYZ5eXlcYgo/pSEHccll8D48XDPPVBWFnY0IpJMDhxQK5gkj/qSsEOHDjW435o1a+jZs2e8woornV81woMP+ickf/ADeOaZsKMRkWShwbslLDNmQHFx6x4zJwd+9rP618+ePZu3336bnJwc0tLS6NatG5mZmRQXF7N161bGjx/Pjh07qKqq4sYbb2TKlCkADBgwgI0bN1JZWcmYMWMYNWoUL730Ev369WP16tV07tw55uc98cQTLFq0iM8//5zTTjuNpUuX0qVLF3bt2sXUqVPZHoxfmJ+fz4UXXkhBQQELFizAzBg6dChLly5tcZmoJawRBg6E2bNh+XIoKgo7GhFJFkrCJJnMnz+fQYMGUVxczP3338/69euZN28eW7duBWDJkiVs2rSJjRs38tBDD7F3795jjrFt2zamTZtGaWkpPXv25Nlnn6338yZMmMCGDRsoKSlh8ODBLF68GIDp06czevRoSkpK2Lx5M9nZ2ZSWljJv3jwKCwspKSlh4cKFrfKd1RLWSLfeCk8/Dd//Prz6KjShLzYRkWZREiZhaajFqq2ce+65R/W59dBDD7EqGMpmx44dbNu2jYyMjKP2GThwIDk5OQAMHz6cd999t97jv/7669x+++2Ul5dTWVnJJZdcAkBhYSEFBQUApKam0qNHDwoKCpg4cSJ9+vQBoHfv3q3yHdUS1kidO/s/ytJSeOSRsKMRkWSgJEySWdeuXY/Mv/DCCzz//POsW7eOkpIShg0bFrNPrk6dOh2ZT01N5eDBg/Ue/9prr+WRRx5hy5Yt3HXXXQ328eWci0v3HUrCmmDsWBgzBu66Cz74IOxoRCTRKQmTZNK9e3cqavplqWP//v306tWLLl268Oabb/Lyyy+3+PMqKirIzMykurqaZcuWHVmem5tLfn4+4B8KOHDgALm5uaxYseLIJdB9+/a1+PNBSViTmMHChb7PsFmzwo5GRBKdkjBJJhkZGYwcOZIvfelL3HLLLUetu/TSSzl48CBDhw7ljjvu4Pzzz2/x591zzz2cd955XHzxxZx55plHli9cuJCioiKGDBnC8OHDKS0tJTs7mzlz5jB69GjOPvtsbrrpphZ/PoA551rlQG1lxIgRbuPGjaHGcNtt8OMfw9/+BiNHhhpKwiovh+pqP3KBiJltcs6NCDuO1tCUOqxbN5gyxT+hLRJvb7zxBoMHDw47jHYtVhk2VH+pJawZ5syBrCy44QY4Tvcl0gyffw6jRsGQIbrsK8nr8GH4+GO1hIkkMiVhzdC1KzzwgO9DZfp0X1lK63nwQf8AxL59cOWVSnQlOVVW+lclYSItM23aNHJyco6annzyybDDAtRFRbNdcQVs2AALFviWm8cfhxSltC32zjtw990wYQJ87Wtw7bUwd64fsUAkmWjcSJHW8eijj4YdQr2UhDWTGdx3nx9fct48f//S4sWQmhp2ZO2Xc74ftpQU3x3IySfD2rW+fEeN8kNIiSQLJWEiiU9JWAuYwb33QseOvtuK6mrfoasG222e3/4Wfv97f6n35JP9socfhvXr4eqrfSe5WVnhxijSVpSEiSQ+XUBrBXfe6Z+W/OUv/T1M1dVhR9T+VFb6++uGDvWvNbp0gZUroaoK8vJUtpI8lISJJD4lYa1k9mzfgrNypb9f7LPPwo6ofZk7F8rKID//2JbEM86ARYvg73/3T6aKJAMlYSKJT0lYK7rpJn/5bPVqf2N5AyMgSC2vvebvAbv+erjwwtjbXHklfPe7cP/98NxzbRufSBiUhIk0rFu3bmGH0GJKwlrZDTf4JyXXrPFP973zTtgRRdvhwzB1KvTqBfPnN7ztgw/CsGEwaRI0MCarSJOY2aVm9g8ze8vMZsdY/1MzKw6mf5pZea11h2qta9XTAyVhIolPt5DHwZQpkJYG110Hp57qn+y7+mp/mbKVBl5PGEuWwLp18NRTxy+b9HT49a/hnHPg61/3IxZ07NgmYUqCMrNU4FHgYqAM2GBmzznnttZs45ybWWv77wPDah3iU+dcTjxiUxImodo0Az4qbt1j9sqB4T+rd/WsWbPo378/3/ve9wCYO3cuZsbatWv56KOPqK6u5t5772XcuHHH/ajKykrGjRsXc7+CggIWLFiAmTF06FCWLl3Krl27mDp1Ktu3bwcgPz+fC+u7NNOK4toS1ogzzE5mtjxY/4qZDYhnPG1p8mTYvh1+9CPYu9e39mRm+suUv/mN7hkD2LMHbr0VLroIvvWtxu0zaBA8+aTvo+0rX/FPo77/fnzjlIR2LvCWc267c+5z4BmgoRr+SuBXbRFYRYV/Artr17b4NJHw5eXlsXz58iPvV6xYweTJk1m1ahWbN2+mqKiIm2++mcYMt5ienh5zv9LSUubNm0dhYSElJSUsXLgQgOnTpzN69GhKSkrYvHkz2dnZcfuetcWtJawxZ5jAt4GPnHOnmVke8BPgG/GKqa317w8//KG/af/VV+EXv/BPUK5aBT17wuWXwxe/CH37wkkn/Xvq29f3P5bobr3V/0eTn+//s2msCRN8H20PPOA7cwXIyYFLL/XTBReohUwarR+wo9b7MuC8WBuaWX9gIFBYa3G6mW0EDgLznXO/ba3AKir82JFN+W2ItJoGWqziZdiwYezevZudO3eyZ88eevXqRWZmJjNnzmTt2rWkpKTw3nvvsWvXLr7whS80eCznHLfddtsx+xUWFjJx4kT69OkDQO/gEkxhYSEFBQUApKam0qNHj/h+2UA8L0ceOcMEMLOaM8zaSdg4YG4wvxJ4xMzMtbdRxY/DzF9CO+ccnzwUFvqEbNUqP1B1LD17+sGru3aFzp19Vw2xXtPSfMKRlnb0fMeO/ilDs2Onmpjqvq8db6z5xryPxTk/1cwD7NrlL0HOng1nnXX8Y9R1yy1w883+pv4//tFPCxb4+8q6d4fcXH/cuuVSe76mY926ZVJ3vrHfvallE+/1xxP28ceM8f8OIYsVZX31Tx6w0jlXeyCtU5xzO83sVKDQzLY4594+5kPMpgBTAE455ZRGBVZRoUuRknwmTpzIypUr+eCDD8jLy2PZsmXs2bOHTZs2kZaWxoABA6hqxFNv9e3nnMMidGYTzySsMWeYR7Zxzh00s/1ABvBh7Y2aU4FFVYcO8OUv+wngk09g926flNSdPvzQr//kE/j0Uz+W4qefHr2sutoPm9QenX463HFH8/dPSfEtYDk5Ppk7cACKiv6dlP3udxp3MsrKy6GNTjYbUgacXOt9FrCznm3zgGm1Fzjndgav283sBfz9YsckYc65RcAigBEjRjTqJFNJmCSjvLw8rr/+ej788ENefPFFVqxYQd++fUlLS6OoqIh//etfjTrO/v37Y+6Xm5vL5ZdfzsyZM8nIyGDfvn307t2b3Nxc8vPzmTFjBocOHeLjjz/mhBNOiOdXBeKbhDXmDLNRZ6HNqcDaiy5dYMAAPzWXcz7ZqEnIqqv/PdW0QtVtjYrVOtXQfGPex4qrbktb7ddTTvGtea3lhBNg3Dg/1Th8GA4ePLpcauYPHz62TOrO1/4uLXlfV7zXH0/Yxwd/qS0CNgCnm9lA4D18ovXNuhuZ2RlAL2BdrWW9gE+cc5+ZWR9gJHBfawV2//3+xEIkmWRnZ1NRUUG/fv3IzMzkqquuYuzYsYwYMYKcnBzOPPPMRh2nvv2ys7OZM2cOo0ePJjU1lWHDhvHUU0+xcOFCpkyZwuLFi0lNTSU/P58LLrggnl8ViG8S1pgzzJptysysA9AD2BfHmBKSmW9h69ChdZOaRJCS4i8/6h4xiSVogb8B+BOQCixxzpWa2d3ARudcTbcTVwLP1LlVYjDwuJkdxj/kNL/OPa8t0r9/ax1JpH3ZsmXLkfk+ffqwbt26mNtVVlbWe4yG9ps0aRKTJk06atlJJ53E6tWrmxFty8QzCWvMGeZzwCT82eVEoDDR7gcTkWhzzq0B1tRZdmed93Nj7PcSMCSuwYlIQotbEtbIM8zFwFIzewvfApYXr3hEREQksWzZsoVrrrnmqGWdOnXilVdeCSmipolrZ63HO8N0zlUBV8QzBhEREUlMQ4YMobi4lTuVbUMatkhEREQAGtURqsTWnLJTEiYiIiKkp6ezd+9eJWLN4Jxj7969pKenN2k/jR0pIiIiZGVlUVZWxp49e8IOpV1KT08nKyurSfsoCRMRERHS0tIYOHBg2GEkFV2OFBEREQmBkjARERGRECgJExEREQmBtbenIMxsD1B7BM8+1BnwO2KiHF+UYwPF11JRjq+psfV3zp0Yr2DaUp06LMr/RqD4WiLKsYHia6mmxFdv/dXukrC6zGyjc25E2HHUJ8rxRTk2UHwtFeX4ohxbW4p6OSi+5otybKD4Wqq14tPlSBEREZEQKAkTERERCUEiJGGLwg7gOKIcX5RjA8XXUlGOL8qxtaWol4Pia74oxwaKr6VaJb52f0+YiIiISHuUCC1hIiIiIu1Ou03CzOxSM/uHmb1lZrPDjqcuM3vXzLaYWbGZbYxAPEvMbLeZvV5rWW8z+4uZbQtee0Usvrlm9l5QhsVmdllIsZ1sZkVm9oaZlZrZjcHySJRfA/FFpfzSzWy9mZUE8f1PsHygmb0SlN9yM+sYRnxhUR3WpFhUf7UsvsjWYUlffznn2t0EpAJvA6cCHYES4Kyw46oT47tAn7DjqBXPRcA5wOu1lt0HzA7mZwM/iVh8c4EfRKDsMoFzgvnuwD+Bs6JSfg3EF5XyM6BbMJ8GvAKcD6wA8oLljwHfDTvWNiwT1WFNi0X1V8vii2wdluz1V3ttCTsXeMs5t9059znwDDAu5JgizTm3FthXZ/E44Olg/mlgfJsGVUs98UWCc+5959zmYL4CeAPoR0TKr4H4IsF5lcHbtGBywH8BK4Plof79hUB1WBOo/mqZKNdhyV5/tdckrB+wo9b7MiL0jxZwwJ/NbJOZTQk7mHqc5Jx7H/wPAegbcjyx3GBmrwXN/aFdbqhhZgOAYfizociVX534ICLlZ2apZlYM7Ab+gm8FKnfOHQw2ieJvOJ5Uh7Vc5H5/MUTi91dblOuwZKy/2msSZjGWRe0xz5HOuXOAMcA0M7so7IDaoXxgEJADvA88EGYwZtYNeBaY4Zw7EGYsscSILzLl55w75JzLAbLwrUCDY23WtlGFSnVY4ovM769GlOuwZK2/2msSVgacXOt9FrAzpFhics7tDF53A6vw/3BRs8vMMgGC190hx3MU59yu4I//MPAEIZahmaXhK4hlzrnfBIsjU36x4otS+dVwzpUDL+DvqehpZh2CVZH7DceZ6rCWi8zvL5ao/f6iXIclc/3VXpOwDcDpwdMJHYE84LmQYzrCzLqaWfeaeeDLwOsN7xWK54BJwfwkYHWIsRyjpnIIXE5IZWhmBiwG3nDOPVhrVSTKr774IlR+J5pZz2C+M/Df+Ps+ioCJwWaR+/uLM9VhLReJ3199ovL7C2KJbB2W9PVX2E8eNHcCLsM/RfE2MCfseOrEdir+aacSoDQK8QG/wjfpVuPPwr8NZAB/BbYFr70jFt9SYAvwGr6yyAwptlH4pubXgOJguiwq5ddAfFEpv6HAq0EcrwN3BstPBdYDbwG/BjqF9fcXUrmoDmt8PKq/WhZfZOuwZK+/1GO+iIiISAja6+VIERERkXZNSZiIiIhICJSEiYiIiIRASZiIiIhICJSEiYiIiIRASZi0e2b2n2b2f2HHISLSVKq/kpuSMBEREZEQKAmTNmNmV5vZejMrNrPHg0FRK83sATPbbGZ/NbMTg21zzOzlYPDWVTWDt5rZaWb2vJmVBPsMCg7fzcxWmtmbZrYs6IUZM5tvZluD4ywI6auLSDun+kviQUmYtAkzGwx8Az8ocA5wCLgK6Apsdn6g4BeBu4JdCoBZzrmh+F6Ta5YvAx51zp0NXIjvpRpgGDADOAvfk/FIM+uNH+4iOzjOvfH9liKSiFR/SbwoCZO2kgsMBzaYWXHw/lTgMLA82OYXwCgz6wH0dM69GCx/GrgoGMuun3NuFYBzrso590mwzXrnXJnzg70WAwOAA0AV8L9mNgGo2VZEpClUf0lcKAmTtmLA0865nGA6wzk3N8Z2DY2jZQ2s+6zW/CGgg3PuIHAu8CwwHvhjE2MWEQHVXxInSsKkrfwVmGhmfQHMrLeZ9cf/DdaMRP9N4G/Ouf3AR2b2H8Hya4AXnXMHgDIzGx8co5OZdanvA82sG9DDObcG39SfE48vJiIJT/WXxEWHsAOQ5OCc22pmtwN/NrMUoBqYBnwMZJvZJmA//r4LgEnAY0EltR2YHCy/BnjczO4OjnFFAx/bHVhtZun4s9CZrfy1RCQJqP6SeDHnGmo9FYkvM6t0znULOw4RkaZS/SUtpcuRIiIiIiFQS5iIiIhICNQSJiIiIhICJWEiIiIiIVASJiIiIhICJWEiIiIiIVASJiIiIhICJWEiIiIiIfh/icJ1dq8xshwAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 720x360 with 2 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "his_dict = history.history\n",
    "loss = his_dict['loss']\n",
    "val_loss = his_dict['val_loss'] \n",
    "\n",
    "epochs = range(1, len(loss) + 1)\n",
    "fig = plt.figure(figsize = (10, 5))\n",
    "\n",
    "# 훈련 및 검증 손실 그리기\n",
    "ax1 = fig.add_subplot(1, 2, 1)\n",
    "ax1.plot(epochs, loss, color = 'blue', label = 'train_loss')\n",
    "ax1.plot(epochs, val_loss, color = 'orange', label = 'val_loss')\n",
    "ax1.set_title('train and val loss')\n",
    "ax1.set_xlabel('epochs')\n",
    "ax1.set_ylabel('loss')\n",
    "ax1.legend()\n",
    "\n",
    "acc = his_dict['acc']\n",
    "val_acc = his_dict['val_acc']\n",
    "\n",
    "# 훈련 및 검증 정확도 그리기\n",
    "ax2 = fig.add_subplot(1, 2, 2)\n",
    "ax2.plot(epochs, acc, color = 'blue', label = 'train_acc')\n",
    "ax2.plot(epochs, val_acc, color = 'orange', label = 'val_acc')\n",
    "ax2.set_title('train and val acc')\n",
    "ax2.set_xlabel('epochs')\n",
    "ax2.set_ylabel('acc')\n",
    "ax2.legend()\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 모델 구성하기 - Conv1D + LSTM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "data = np.sin(2 * np.pi * 0.05 + np.arange(0, 500)) + np.random.random(500)\n",
    "plt.figure(figsize = (10, 5)) \n",
    "plt.plot(np.arange(0, 367), data[:367], 'bo-', color = 'blue', alpha = 0.2) # 학습용 데이터\n",
    "plt.plot(np.arange(367, 500), data[367:], 'ro-', color = 'orange') # 테스트용 데이터\n",
    "\n",
    "def data_gen(data, n):\n",
    "    x_train = []\n",
    "    y_train = []\n",
    "    \n",
    "    for i in range(len(data)):\n",
    "        x = data[i:(i + n)]\n",
    "        if (i + n) < len(data):\n",
    "            x_train.append(x)\n",
    "            y_train.append(data[i + n])\n",
    "        else:\n",
    "            break\n",
    "            \n",
    "    return np.array(x_train), np.array(y_train)\n",
    "\n",
    "n = 10\n",
    "\n",
    "x_train, y_train = data_gen(data, n)\n",
    "x_train = x_train.reshape(-1, n, 1)\n",
    "y_train = y_train.reshape(-1, 1)\n",
    "\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "x_train, x_test, y_train, y_test = train_test_split(x_train, y_train, test_size = 0.25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tensorflow.keras.models import Sequential\n",
    "from tensorflow.keras.layers import Dense, Embedding, Conv1D, MaxPooling1D, GlobalMaxPooling1D\n",
    "from tensorflow.keras.layers import LSTM\n",
    "\n",
    "model = Sequential()\n",
    "model.add(Conv1D(32, 3, activation = 'relu', input_shape = (10, 1)))\n",
    "model.add(MaxPooling1D(2))\n",
    "model.add(Conv1D(32, 3, activation = 'relu'))\n",
    "# LSTM을 혼합하여 모델을 구성합니다.\n",
    "model.add(LSTM(32, dropout = 0.2, recurrent_dropout = 0.2))\n",
    "model.add(Dense(1))\n",
    "\n",
    "model.compile(optimizer='adam',\n",
    "             loss = 'mse',\n",
    "             metrics = ['mse'])\n",
    "\n",
    "# 학습을 진행합니다.\n",
    "model.fit(x_train, y_train, epochs = 100, batch_size = 12)\n",
    "\n",
    "pred = model.predict(x_test)\n",
    "\n",
    "# 결과값을 그립니다.\n",
    "train_axis = np.arange(0, len(y_train))\n",
    "pred_axis = np.arange(len(y_train), len(y_train) + len(pred))\n",
    "train_shape= y_train.shape[0]\n",
    "test_shape = y_test.shape[0]\n",
    "pred_shape = pred.shape[0]\n",
    "\n",
    "plt.figure(figsize=(20,5))\n",
    "plt.plot(train_axis, y_train.reshape(train_shape,), 'o-')\n",
    "plt.plot(pred_axis, pred.reshape(pred_shape,), 'o-', color='orange', label='pred')\n",
    "plt.plot(pred_axis, y_test.reshape(test_shape,), 'o-', color='blue', alpha=0.2, label='ground-truth')\n",
    "plt.legend()\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "keras_test",
   "language": "python",
   "name": "keras_test"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
